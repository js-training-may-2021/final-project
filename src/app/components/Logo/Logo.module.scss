@import "@scss/variables.scss";
@import "@scss/mixins.scss";

.logo {
  @include font-size(24, 28);
  @include d-flex-standart;

  width: 150px;
  padding: 0 10px;

  img {
    width: 100%;
  }

  @media screen and (max-width: $breakpoint4) {
    width: 100px;
  }
}
