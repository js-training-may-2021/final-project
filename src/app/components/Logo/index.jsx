import React from 'react';
import { Link } from 'react-router-dom';

import { loc } from '@utils/languageUtils';

import styles from './Logo.module.scss';

const Logo = () => (
  <Link to="/">
    <div className={styles.logo}>
      <img src="/images/icons/logo.png" alt={loc('altLogo')} />
    </div>
  </Link>

);

export default Logo;
