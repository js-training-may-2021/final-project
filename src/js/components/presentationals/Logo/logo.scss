@import '../../../../styles/variables';

.logo {  
  margin: 0 40px;
  font-size: 2rem;
  font-weight: 500;
}

@media (max-width: $mobileWidth) {
  .logo {
    margin: 0 20px 5px;
    font-size: 1.5rem;
  }
}
