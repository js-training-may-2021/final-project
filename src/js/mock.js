export const pokemons = [
  {
    name: 'bulbasaur',
    id: 1,
    isCaught: false,
    catchDate: null,
  },
  {
    name: 'ivysaur',
    id: 2,
    isCaught: true,
    catchDate: '23 June 2021',
  },
  {
    name: 'venusaur',
    id: 3,
    isCaught: true,
    catchDate: '23 June 2021'
  },
  {
    name: 'charmander',
    id: 4,
    isCaught: false,
    catchDate: null,
  },
  {
    name: 'charmeleon',
    id: 5,
    isCaught: true,
    catchDate: '23 June 2021'
  },
  {
    name: 'charizard',
    id: 6,
    isCaught: true,
    catchDate: '23 June 2021'
  },
  {
    name: 'squirtle',
    id: 7,
    isCaught: false,
    catchDate: null,
  },
  {
    name: 'wartortle',
    id: 8,
    isCaught: true,
    catchDate: '23 June 2021'
  },
  {
    name: 'blastoise',
    id: 9,
    isCaught: true,
    catchDate: '23 June 2021'
  },
  {
    name: 'caterpie',
    id: 10,
    isCaught: false,
    catchDate: null,
  },
  {
    name: 'metapod',
    id: 11,
    isCaught: false,
    catchDate: null,
  }];  

export const uncaughtPokemon = pokemons[0];
export const caughtPokemon = pokemons[1];
