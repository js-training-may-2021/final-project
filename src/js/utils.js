import {createMemoryHistory} from 'history';
import configureStore from 'redux-mock-store';

export const POKEMONS_PER_PAGE = 10;

let currentPage = 1;

export const getCurrentPage = () => {
	return currentPage++;
};

export const AppRoute = {
  MAIN: '/',
  DETAIL: '/detail/:id',
  CAUGHT: '/caught'
};

const monthNames = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];

export const getCatchDate = () => {
  const date = new Date();

  const day = date.getDate();
  const month = monthNames[date.getMonth()];
  const year = date.getFullYear();

  return `${day} ${month} ${year}`
};

export const noop = () => {};
export const history = createMemoryHistory();
const mockStore = configureStore();
export const testStore = mockStore({});
