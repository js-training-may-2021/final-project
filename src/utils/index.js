export const firstLetterToUpperCase = (str) => {
  if (str === undefined) {
    return str;
  } else {
    return str
      .toLowerCase()
      .split(" ")
      .map((s) => s.charAt(0).toUpperCase() + s.substring(1))
      .join(" ");
  }
};

export const formatDate = (dateInput) => {
  return dateInput.toLocaleString("ru-RU");
};

export const padToThree = (number) => number <= 999 ? `00${number}`.slice(-3) : number;

export const handleErrors = (response) => {
  if (!response.ok) {
    throw Error(`${response.status}: ${response.statusText}`);
  }
  return response;
};
