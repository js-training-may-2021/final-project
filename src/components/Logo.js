import React from 'react';
import logo from '../assets/images/pokemonLogo.svg';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

const Logotype = styled.img`
    @media (max-width: 620px) {
        max-width: 140px;
    }
    @media (max-width: 370px) {
        max-width: 110px;
    }
`
function Logo() {
    return (
        <div>
            <Link to="/">
                <Logotype src={logo} alt={'logo'}/>
            </Link>
        </div>
    );
}

export default Logo