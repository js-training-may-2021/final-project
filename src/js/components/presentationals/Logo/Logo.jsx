import React from 'react';
import {Link} from 'react-router-dom';
import {AppRoute} from '../../../utils';
import './logo.scss';

const Logo = () => {
  return (    
    <div className="logo">
      <Link className="logo__link" to={AppRoute.MAIN}>POKEDEX</Link>
    </div>
  );
};

export default React.memo(Logo);