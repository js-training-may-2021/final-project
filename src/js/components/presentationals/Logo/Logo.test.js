import React from 'react';
import renderer from 'react-test-renderer';
import {Router} from 'react-router-dom';
import Logo from './Logo.jsx';
import {history} from '../../../utils.js';

it(`Render Logo`, () => {
  const tree = renderer
    .create(
      <Router history={history}>
        <Logo />
      </Router>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
