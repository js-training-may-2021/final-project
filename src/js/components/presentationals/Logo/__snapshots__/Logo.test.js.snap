// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Logo 1`] = `
<div
  className="logo"
>
  <a
    className="logo__link"
    href="/"
    onClick={[Function]}
  >
    POKEDEX
  </a>
</div>
`;
